src/main/java/org/folio/search/configuration/OpensearchRestClientConfiguration.java

```diff
@@ -112,24 +128,59 @@ static class DefaultRestClientBuilderCustomizer implements RestClientBuilderCust
     @Override
     public void customize(HttpAsyncClientBuilder builder) {
       builder.setDefaultCredentialsProvider(new PropertiesCredentialsProvider(this.properties));
+      builder.setConnectionManager(getPoolingAsyncClientConnectionManager());
+      if (properties.isElasticsearchServer()) {
+        builder.addRequestInterceptorFirst((HttpRequest request, EntityDetails entityDetails, HttpContext context) -> {
+          var uri = request.getRequestUri();
+          if (uri.contains("cluster_manager_timeout")) {
+            var newUri = uri.replaceAll("[&?]?cluster_manager_timeout=[^&]*", "");
+            request.setPath(newUri);
```

Comment:
Could you extract this to seaprate bean with conditional on properties value? In the configuration - inject list of interceptors and use them in this method.
Then you will be able to write clean unit tests for the interceptor.